scripts/apitypings/main.go

```diff
@@ -691,7 +696,16 @@ func (g *Generator) typescriptType(ty types.Type) (TypescriptType, error) {
 			if err != nil {
 				return TypescriptType{}, xerrors.Errorf("array: %w", err)
 			}
-			return TypescriptType{ValueType: underlying.ValueType + "[]", AboveTypeLine: underlying.AboveTypeLine}, nil
+			var genValue = ""
+			if underlying.GenericValue != "" {
+				genValue = underlying.GenericValue + "[]"
+			}
+			return TypescriptType{
+				ValueType:     underlying.ValueType + "[]",
+				GenericValue:  genValue,
+				AboveTypeLine: underlying.AboveTypeLine,
+				GenericTypes:  underlying.GenericTypes,
+			}, nil
```

Comment:
The only reason this did not exist is because we did not need it yet. It's a trivial addition.